checkita-core/src/main/scala/org/checkita/dqf/readers/SourceReaders.scala

```diff
@@ -41,6 +41,7 @@ object SourceReaders {
      */
     protected def toSource(config: T, df: DataFrame, checkpoint: Option[Checkpoint] = None)
                           (implicit settings: AppSettings): Source = {
+      if (config.persist.nonEmpty) df.persist(config.persist.get)
```

Comment:
Need to refactor this function, to persist only batch sources. Add ReadMode to function parameters and do persist only for batch mode.